include/hffix.hpp

```diff
@@ -532,13 +538,14 @@ class message_writer {
             throw std::logic_error("hffix message_writer.push_back_trailer called before message_writer.push_back_header");
         }
 
-        size_t const len = next_ - (body_length_ + 7);
-        body_length_[0] = '0' + (len / 100000) % 10;
-        body_length_[1] = '0' + (len / 10000) % 10;
-        body_length_[2] = '0' + (len / 1000) % 10;
-        body_length_[3] = '0' + (len / 100) % 10;
-        body_length_[4] = '0' + (len / 10) % 10;
-        body_length_[5] = '0' + len % 10;
+        size_t const len = next_ - (body_length_ + BodyLengthDigits + 1);
+        for (std::size_t i = 0; i < BodyLengthDigits; ++i) {
+            body_length_[i] = '0' + (len / details::pow(10, BodyLengthDigits - i - 1)) % 10;
+        }
```

Comment:
Since the count is known at compile time, I would expect this loop to get unrolled, and there are no data dependencies between the iterations so it should be really friendly to parallelization and instruction reordering.